tests/test_types.py

```diff
@@ -0,0 +1,154 @@
+"""
+:Description: Unit tests for the types module
+"""
+
+from __future__ import annotations
+
+import pytest
+
+from conda_recipe_manager.types import MessageCategory, MessageTable
+
+
+@pytest.mark.parametrize(
+    "messages_and_contents,expected",
+    [
+        (
+            [
+                (MessageCategory.WARNING, "Test warning message"),
+            ],
+            ("0 errors and 1 warning were found."),
+        ),
+        (
+            [
+                (MessageCategory.ERROR, "Test error message"),
+            ],
+            ("1 error and 0 warnings were found."),
+        ),
+        (
+            [
+                (MessageCategory.WARNING, "Test warning message"),
+                (MessageCategory.ERROR, "Test error message"),
+            ],
+            ("1 error and 1 warning were found."),
+        ),
+        (
+            [
+                (MessageCategory.WARNING, "Test warning message"),
+                (MessageCategory.ERROR, "Test error message"),
+                (MessageCategory.ERROR, "Another test error message"),
+            ],
+            ("2 errors and 1 warning were found."),
+        ),
+        (
+            [
+                (MessageCategory.WARNING, "Test warning message"),
+                (MessageCategory.ERROR, "Test error message"),
+                (MessageCategory.EXCEPTION, "Test exception message"),
+            ],
+            ("1 error and 1 warning were found."),
+        ),
+        (
+            [
+                (MessageCategory.EXCEPTION, "Test exception message"),
+            ],
+            ("0 errors and 0 warnings were found."),
+        ),
+    ],
+)
+def test_message_table_add(messages_and_contents: list[tuple[MessageCategory, str]], expected: str) -> None:
+    """
+    Tests the addition of messages to the message table and checks the expected output of warnings and errors
+    contained in the messaging object.
+
+    param messages_and_contents: Message category and message content
```

Comment:
```suggestion
    :param messages_and_contents: Message category and message content
```

Reply:
Actually all of these are like this....

Reply:
I have a pylint rule to enforce doc strings but it seems to be flaky/inconsistent.